### PR TITLE
ChefDK 0.5.0.rc omnibus release

### PR DESCRIPTION
**Pull request by tyler-ball**
\cc @chef/client-core 

Before 0.5.0 can be released, I want to pull in the released versions of the test kitchen dependencies and the final (non-rc) version of ChefDK.

### CODE REVIEW

**Comment by jdmundrawala** [comment]
Looks good, might want to check if Foodcritic can be bumped. You'll also need to wait for the chef 12.2.0 gem to be published

File: config/software/chefdk.rb [hunk 1]
```diff
@@ -15,7 +15,7 @@
 #
 
 name "chefdk"
-default_version "master"
+default_version "0_5_release"
```
Comment (danielsdeleo) [inline config/software/chefdk.rb]: The tags are usually just the version numbers, did you create a new branch?

**Comment by danielsdeleo** [comment]
Also as long as the Chef 12.0.0 tag is on github, we can build this, though that'll mean that when we `bundle install` chef-dk it'll pull in the RC from rubygems.

**Comment by danielsdeleo** [comment]
:+1: all looks sane

File: config/software/chefdk.rb (hunk 1 above)
Comment (tyler-ball) [inline config/software/chefdk.rb]: Yeah, this was a branch, but I just taggd `0.5.0.rc.1`

**Comment by tyler-ball** [comment]
@jdmundrawala foodcritic is at the latest, v4.0.0

**Comment by schisamo** [comment]
Let's squash these commits down, rebase on master and get this :ship:'d!

**Pull request merged**